cfpb/cfgov-refresh
# Restore missing migration testing

See discussion in https://github.com/cfpb/cfgov-refresh/pull/3078.

Since we're not running migrations in Travis our missing migration unit test no longer works in Travis. This is an attempt to perform the same test by copying the migration test code into a script that can be run by the `runscript` management command in a tox environment.

The script can be removed and replaced by `makemigrations --check` in Django 1.10+.

The output in Travis should appear like this for failure:
```
Missing migrations found:
Migrations for 'v1':
  0070_cfgovpage_changed.py:
    - Add field changed to cfgovpage
```

## Testing

1. Run `tox -e missing-migrations` and see that everything suceeds.
2. Change a model in a trivial way for testing.
3. Run `tox -e missing-migrations` and see that it fails and prints the expected migration it would've created.
4. Revert your change to the model

## Checklist
